zebra-chain/src/sprout/tree.rs

```diff
@@ -10,11 +10,69 @@
 //!
 //! A root of a note commitment tree is associated with each treestate.
 #![allow(clippy::unit_arg)]
+#![allow(dead_code)]
 
-use std::fmt;
+use std::{collections::VecDeque, fmt};
 
+use byteorder::{BigEndian, ByteOrder};
+use lazy_static::lazy_static;
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
+use sha2::digest::generic_array::GenericArray;
+
+use super::commitment::NoteCommitment;
+
+const MERKLE_DEPTH: usize = 29;
+
+/// MerkleCRH^Sprout Hash Function
+///
+/// MerkleCRH^Sprout(layer, left, right) := SHA256Compress(left || right)
+///
+/// `layer` is unused for Sprout but used for the Sapling equivalent.
+///
+/// https://zips.z.cash/protocol/protocol.pdf#merklecrh
```

Comment:
same as the other review, I think it would help a lot to give examples and simple explanations of how to use these more crypto heavy functions, even if that's just explaining how it's used in the `From` impl and giving an example that uses the `From` impl